### ISSUE TITLE
Possible improvement for 3.0.0 breaking change

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1174]
Reported by downstream here:

https://github.com/microsoft/LightGBM/issues/5228

Slightly simpler reprex:

```
lintr::lint(text = "print(1)", linters = lintr::assignment_linter)
# Error: `fun` must be a function taking exactly one argument.
# In addition: Warning message:
#  The use of linters of class 'function' was deprecated in lintr version 2.0.1.9001. Use linters classed as 'linter' (see ?Linter) instead.
```

Can we not do `fun()` for the user in this release after issuing "please use a linter" warning?

Also, I'm wondering if the warning message can't be improved:

> The use of linters of class 'function' was deprecated in lintr version 2.0.1.9001. Use linters classed as 'linter' (see ?Linter) instead.

I think the most likely scenario where a user encounters this warning is just like the above -- they've got a script using the un-called form, or they've grown used to supplying it themselves interactively. So we should add this to the end, right?

> For example, if you supplied `lintr::assignment_linter`, please use `lintr::assignment_linter()` instead.

**Comment by AshesITR** [comment]
That was supposed to happen already. Good catch!

**Comment by MichaelChirico** [comment]
completely untested suspicion is it's because assignment_linter has arguments and we've only tested it on unparameterized linters?
